functions/console_snippets/http/http_test.go

```diff
@@ -0,0 +1,72 @@
+package p
+
+import (
+	"bufio"
+	"bytes"
+	"log"
+	"net/http"
+	"net/http/httptest"
+	"os"
+	"strings"
+	"testing"
+)
+
+func TestHelloWorld(t *testing.T) {
+	tests := []struct {
+		name string
+		want string
+		data string
+	}{
+		{
+			name: "valid",
+			want: "Greetings, Ocean!",
+			data: `{"message": "Greetings, Ocean!"}`,
+		},
+		{
+			name: "empty",
+			want: "Hello, World!",
+			data: "",
+		},
+		{
+			name: "invalid",
+			want: "Hello, World!",
```

Comment:
I'm not sure why we would intentionally want data="" and data="not-valid-JSON" to produce the same result.

A "Data not valid." result or something else would seem fine.